src/containers/CheckoutPage/CheckoutPage.js

```diff
@@ -138,61 +153,122 @@ export class CheckoutPageComponent extends Component {
     this.setState({ pageData: pageData || {}, dataLoaded: true });
   }
 
-  handleSubmit(values) {
-    if (this.state.submitting) {
-      return;
-    }
-    this.setState({ submitting: true });
-
-    const cardToken = values.token;
-    const initialMessage = values.message;
+  handlePaymentIntent(handlePaymentParams) {
     const {
-      history,
-      sendOrderRequest,
-      sendOrderRequestAfterEnquiry,
-      speculatedTransaction,
-      dispatch,
+      // onRequestPayment,
+      onHandleCardPayment,
+      // onConfirmPayment,
     } = this.props;
 
+    const fnRequestPayment = parameters => {
+      // parameters should be { listingId, bookingStart, bookingEnd }
+      const { stripePaymentIntentClientSecret, stripePaymentIntentId } = handlePaymentParams;
+      return stripePaymentIntentClientSecret
+        ? Promise.resolve({ stripePaymentIntentClientSecret, stripePaymentIntentId })
+        : Promise.resolve(parameters); // : onRequestPayment(parameters);
+    };
+
+    const fnHandleCardPayment = parameters => {
+      // parameters should be { stripePaymentIntentClientSecret, stripePaymentIntentId }
+      const { stripe, card, cardholderName, email, paymentIntent } = handlePaymentParams;
+      const params = {
+        ...parameters,
+        stripe,
+        card,
+        paymentParams: {
+          payment_method_data: {
+            billing_details: {
+              name: cardholderName,
+              email,
+              // address: {
+              //   city: null,
+              //   country: null,
+              //   line1: null,
+              //   line2: null,
+              //   postal_code: null,
+              //   state: null
+              // },
+            },
+          },
+        },
+      };
+      const hasPaymentIntentSucceeded = paymentIntent && paymentIntent.status === 'succeeded';
+      return hasPaymentIntentSucceeded
+        ? Promise.resolve(paymentIntent)
+        : onHandleCardPayment(params);
+    };
+
+    const fnConfirmPayment = parameters => {
+      const { message } = handlePaymentParams;
+      const params = { ...parameters, message };
+      return Promise.resolve(params); // onConfirmPayment(params);
+    };
+
```

Comment:
Adding a comment to clarify what magic is happening here and why might be useful.